src/dmd/dsymbolsem.d

```diff
@@ -4397,15 +4397,36 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
             // Treat the remaining entries in baseclasses as interfaces
             // Check for errors, handle forward references
+            bool multiClassError = false;
+            
             for (size_t i = (cldec.baseClass ? 1 : 0); i < cldec.baseclasses.dim;)
             {
                 BaseClass* b = (*cldec.baseclasses)[i];
                 Type tb = b.type.toBasetype();
                 TypeClass tc = (tb.ty == Tclass) ? cast(TypeClass)tb : null;
                 if (!tc || !tc.sym.isInterfaceDeclaration())
                 {
-                    if (b.type != Type.terror)
+                    // It's a class
+                    if (tc)
+                    {
+                        if (!multiClassError)
+                        {
+                            cldec.error("D does not support multiple class inheritance." ~
```

Comment:
You might  want to use `error(cldec.loc, "D does not..")` instead of `cldec.error()` because the `class `diag18574.Test` prefix that `cldec.error` adds to the error message makes it read weird.

Reply:
I See what you mean, but it is nice to have the symbol for where the source location doesn't correspond well to the error (cf. templates mixins), what about `Error: class diag18574.Test multiple class inheritance is not supported. Use ...`?

Reply:
I think if there was simply a colon after the prefix, it would read a lot better.  What about `error(cldec.loc, "%s: D does not support...,", cldec.toPrettyChars())`?  It's on my list to fix the prefix oddity (https://github.com/dlang/dmd/projects/3#card-6750742).